GitUI/UserControls/RevisionGrid/Columns/NotesColumnProvider.cs

```diff
@@ -0,0 +1,42 @@
+using GitCommands;
+using GitExtUtils.GitUI;
+using GitUIPluginInterfaces;
+
+namespace GitUI.UserControls.RevisionGrid.Columns;
+
+internal sealed class NotesColumnProvider : ColumnProvider
+{
+    private readonly RevisionGridControl _grid;
+
+    public NotesColumnProvider(RevisionGridControl grid) : base("Notes")
+    {
+        _grid = grid;
+
+        var column = new DataGridViewTextBoxColumn
+        {
+            AutoSizeMode = DataGridViewAutoSizeColumnMode.None,
+            HeaderText = Name,
+            ReadOnly = false,
+            SortMode = DataGridViewColumnSortMode.NotSortable,
+            Width = DpiUtil.Scale(50),
+            MinimumWidth = DpiUtil.Scale(25),
+        };
+        Column = column;
+    }
+
+    public override void OnCellPainting(DataGridViewCellPaintingEventArgs e, GitRevision revision, int rowHeight, in CellStyle style)
+    {
+        if (FirstLine(revision.Notes) is string str)
+        {
+            _grid.DrawColumnText(e, str, style.NormalFont, style.ForeColor, e.CellBounds);
+        }
+    }
+
+    private static string? FirstLine(string? str) => str?.IndexOf('\n') is { } idx and > 0 ? str[..idx] : str;
```

Comment:
ditto

Reply:
What's a `str`? Same with `idx`. Why would anyone need to guess?

Reply:
Why guess? It`s in the very same line.
Plus: https://github.com/abbrcode/abbreviations-in-code

Reply:
Much clearer IMO - presuming `\r` cannot appear in the `text`:
```suggestion
    private static string? FirstLine(string? text) => text?.IndexOf('\n') is int eolIndex and > 0 ? text[..eolIndex] : text;
```
(Basically, `is { }` is terrible syntax.)